src/Nethermind/Nethermind.Core/Collections/SpanDictionary.cs

```diff
@@ -51,7 +51,7 @@ public SpanDictionary(int capacity, ISpanEqualityComparer<TKey> comparer)
         }
 
         public SpanDictionary(IDictionary<TKey[], TValue> dictionary, ISpanEqualityComparer<TKey> comparer) :
-            this(dictionary != null ? dictionary.Count : 0, comparer)
+            this(dictionary is not null ? dictionary.Count : 0, comparer)
```

Comment:
one thing about this class - this was copied directly from .net source, not sure if we want to keep it as close as possible as it can work with generic types or are fine with changes here

Reply:
Are object types not T types or C# would fail to compile on the `is null` test